packages/kamino-sdk/src/rebalance_methods/takeProfitRebalance.ts

```diff
@@ -78,19 +78,44 @@ export function getDefaultTakeProfitRebalanceFieldsInfos(price: Decimal): Rebala
   return getTakeProfitRebalanceFieldsInfos(lowerPrice, upperPrice, price);
 }
 
-export function deserializeTakeProfitRebalanceFromOnchainParams(
+export function readTakeProfitRebalanceParamsFromStrategy(
   tokenADecimals: number,
   tokenBDecimals: number,
   rebalanceRaw: RebalanceRaw
-): RebalanceFieldInfo[] {
+) {
```

Comment:
I think this breaks a lot the consistency and the flow for the UI. I think that UI calls this,the UI gets a list of `RebalanceFieldInfo` that can be directly rendered in the UI and that can be reused to call all the other function. I think it would be great to use `RebalanceFieldInfo`everywhere we can as it is similar enough to a map, and the `enabled` just represents if that field is directly editable or calculated and can be ignored, depending on the need (it is mostly used by UI). 
Same for all rebalancing strats

Reply:
Just so that we are on the same page: this change is backwards compatible, I only added two more functions to each strategy type.

The UI will just have to display these using a tooltip or table or some generic list. It will not be able to reuse the same components as the ones used for editing the strategy (those are made for writing, these two endpoints are displayed as read only / additional info). There is no need to have field type or if it's enabled, and to have these two endpoints return more data than necessary for the sake of using the same struct in multiple places. 

What do you think?

Reply:
I see your point but reusing the `RebalanceFieldInfo` type will be nicer I think, as we won't need a new field, the UI already has the code to interpret it and if we want to use this for other scopes we will need to write adapters between the types